packages/cli/src/databases/dsl/Table.ts

```diff
@@ -0,0 +1,82 @@
+import type { TableForeignKeyOptions, TableIndexOptions } from 'typeorm';
+import { Table, QueryRunner } from 'typeorm';
+import LazyPromise from 'p-lazy';
+import { Column } from './Column';
+
+abstract class TableOperation<R = void> extends LazyPromise<R> {
+	abstract execute(queryRunner: QueryRunner): Promise<R>;
+
+	constructor(
+		protected tableName: string,
+		protected prefix: string,
+		queryRunner: QueryRunner,
+	) {
+		super((resolve) => {
+			void this.execute(queryRunner).then(resolve);
+		});
+	}
+}
+
+export class CreateTable extends TableOperation {
+	private columns: Column[] = [];
+
+	private indicesOn = new Set<TableIndexOptions>();
+
+	private foreignKeys = new Set<TableForeignKeyOptions>();
+
+	withColumns(...columns: Column[]) {
+		this.columns.push(...columns);
+		return this;
+	}
+
+	get withTimestamps() {
+		this.columns.push(
+			new Column('createdAt').timestamp(3).notNull.default('NOW()'),
+			new Column('updatedAt').timestamp(3).notNull.default('NOW()'),
+		);
+		return this;
+	}
+
+	withIndexOn(columnName: string | string[], isUnique = false) {
+		const columnNames = Array.isArray(columnName) ? columnName : [columnName];
+		this.indicesOn.add({ columnNames, isUnique });
+		return this;
+	}
```

Comment:
`isUnique` cannot be specified for each individual column when passing multiple? Maybe `withIndexOn` should accept a single column name. The user can always chain more calls.

Reply:
you can have a compound index that's also unique. one such example is the index on `scope`+`name` on the `role` table.